packages/ui-kit/src/components/Leaderboard/Leaderboard.tsx

```diff
@@ -187,6 +163,37 @@ export const LeaderboardComponent = React.forwardRef<HTMLDivElement, Leaderboard
           plugins: [customCanvasBackgroundColor, customChartLabelsPlugin]
         }
 
+        if (chartRef.current) {
+          const customConfig = chartConfigProps?.(config)
```

Comment:
Could you please share a bug scenario?

We should be careful here. The previous logic was as follows:
1. Create a chart config on the component mount.
2. Modify it via `chartConfigProps`.
3. Initialize the chart with the config.
4. On any further re-renders, modify the existing config.

I added only step 2, but the other steps make sense to me because we don't know for sure how Chart.js modifies the config during the initialization. We don't want to break the internal logic by overriding the entire config on each re-render. Can we keep it? By changing the position of:
```
        if (chartConfigProps) {
          config = chartConfigProps(config)
        }
```
Please share the bug scenario for more accurate suggestions.

Reply:
Sure, here is a demo of the bug

https://github.com/propeldata/ui-kit/assets/84721399/19e789b2-7622-4aa4-9a03-19f907785175

Currently charts are not following step 4, further re-renders are overriding config to the default configuration, in the example above I changed the colors using `chartConfigProps` and after triggering a re-render (by clicking dev tools then the page which triggers react-query) the colors turn to the default ones, this was because the if statement that updates the chart instead of creating a new instance was not receiving the `chartConfigProp` built config object.

> We don't want to break the internal logic by overriding the entire config on each re-render

Not sure why would this break the internal logic as we are only using Chart.js's api to change options, can you provide an example please?

Reply:
I have two suggestions here:
Option 1. Preferable. Remove all visual configurations from the `update` part. Update only the data.
Option 2. Remove `update` part and just re-create (via `destroyChart`) the chart on re-render. We may need to do some refactoring to avoid unnecessary re-renders.

Wdyt?

Reply:
I think we can proceed with option 1, that would fix the bug, however, that removes the possibility of updating styles based on state.

I wouldn't go for option 2 as that could cause flickering issues.

Why do you think it's dangerous to enable updating chart.js config?

Reply:
> that removes the possibility of updating styles based on state.

Do you mean, for example, switching between light/dark modes? Or clicking through different color palettes?

Reply:
> Do you mean, for example, switching between light/dark modes? Or clicking through different color palettes?

Yes, this would disable switching between light/dark mode for example